### PR TITLE
Produce x86_64 & ARM64 fedora container images

### PR DESCRIPTION
**Pull request by cevich**
**Depends on:** #380

At the time of this commit, podman's Makefile has a target to allow
validating code changes locally (`validatepr`).  However, it's based
on a bespoke image completely unassociated with the image used in CI.
This can easily lead to a situation where validation passes in the local
environment but fails in CI.  Support the podman `validatepr` target
use of `quay.io/libpod/fedora_podman:latest` images by performing
a manifest-list build that includes `arm64` (a.k.a. `aarch64`).

The trade-off here is image build-time, since emulation is
extremely slow (over an hour).  Therefore, the `container_images` CI
task has also been removed as a dependency from `base_images` CI task,
allowing them to run in parallel.

Note: This will not impact pulling the image, since the client always
only pulls the layers necessary for the indicated architecture.

Ref: [Podman's validatepr Makefile target](https://github.com/containers/podman/blob/5f069d874297358f629454a2a4a0b96c15a81836/Makefile#L325-L340)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -294,80 +305,87 @@ fedora_podman:  ## Build Fedora podman development container
...
-		--security-opt seccomp=unconfined \
+# N/B: This assumes that qemu_user_static is present and installed into the kernel or (as root)
+# podman run --rm --privileged mirror.gcr.io/multiarch/qemu-user-static:latest --reset -p yes
+# must been executed (e.g. on CentOS).
+$(_TEMPDIR)/%_podman.iid: podman/Containerfile podman/setup.sh $(wildcard base_images/*.sh) $(_TEMPDIR) $(wildcard cache_images/*.sh)
+	  podman build --manifest=$*_podman:$(call err_if_empty,_IMG_SFX) \
+		--net=host --ipc=host --pid=host \
+    --cgroupns=host --security-opt label=disable \
+    --security-opt seccomp=unconfined \
+	  --platform=linux/amd64,linux/arm64 \
```
Comment (cevich) [inline Makefile:316]: note to me: Fix the gawdawful indenting.  Also, probably don't need `--net=host --ipc=host --pid=host --cgroupns=host`.

**Comment by edsantiago** [comment]
- 20240820t151015z-f40f39d13
- 20240805t173227z-f40f39d13 &#x21D1; (built in #371)

|  |*debian*|*prior-fedora*|*fedora*|*fedora-aws*|*rawhide*|
|----|----|----|----|----|----|
|*base*         | 13.5            | 39-1.5          | Generic         | ?               | 42-0            |
|               |                |                |                |                | 41-0&nbsp;&#x21D1; |
|*kernel*       | 6.10.4-1        | 6.5.6-300       | 6.8.5-301       | 6.8.5-301       | 6.8.5-301       |
|               | 6.9.12-1&nbsp;&#x21D1; | 6.9.12-100&nbsp;&#x21D1; | 6.9.12-200&nbsp;&#x21D1; | 6.9.12-200&nbsp;&#x21D1; |                |
|*grub2-common* | 2.12-5          | 2.06-121        | 2.06-123        | 2.06-123        | 2.12-4          |
|               |                |                |                |                | 2.06-124&nbsp;&#x21D1; |
|*aardvark-dns* | 1.9.0-2         | 1.12.1-1        | 1.12.1-1        | 1.12.1-1        | 1.12.1-1        |
|               | 1.6.0-3&nbsp;&#x21D1; | 1.11.0-1&nbsp;&#x21D1; |                | 1.11.0-3&nbsp;&#x21D1; |                |
|*netavark*     | 1.9.0-4         | 1.12.1-1        | 1.12.2-1        | 1.12.1-1        | 1.12.2-1        |
|               | 1.6.0-2.1&nbsp;&#x21D1; | 1.11.0-1&nbsp;&#x21D1; | 1.12.1-1&nbsp;&#x21D1; | 1.11.0-3&nbsp;&#x21D1; | 1.11.0-3&nbsp;&#x21D1; |
|*buildah*      | 1.37.1+ds1-2    | 1.37.0-1        | 1.37.1-1        | 1.37.0-1        | 1.37.1-1        |
|               | 1.35.3+ds1-3&nbsp;&#x21D1; | 1.36.0-1&nbsp;&#x21D1; | 1.37.0-1&nbsp;&#x21D1; | 1.36.0-1&nbsp;&#x21D1; | 1.37.0-1&nbsp;&#x21D1; |
|*containers-common*| ?               | 1-99            | 0.60.1-1        | 0.60.0-1        | 0.60.1-1        |
|               |                |                | 0.60.0-1&nbsp;&#x21D1; |                | 0.60.0-1&nbsp;&#x21D1; |
|*crun*         | 1.16.1-1        | 1.15-1          | 1.15-1          | 1.15-1          | 1.15-2          |
|               | 1.15-1&nbsp;&#x21D1; |                |                |                |                |
|*docker-ce*    | 5:27.1.2-1\~debian.12\~bookworm | ?               | ?               | ?               | ?               |
|               | 5:27.1.1-1\~debian.12\~bookworm&nbsp;&#x21D1; |                |                |                |                |
|*golang*       | 2:1.22\~3       | 1.21.12-1       | 1.22.6-1        | 1.22.6-1        | 1.23.0-2        |
|               |                |                | 1.22.5-1&nbsp;&#x21D1; | 1.22.5-1&nbsp;&#x21D1; | 1.23\~rc2-1&nbsp;&#x21D1; |
|*gvisor-tap-vsock*| ?               | 0.7.4-1         | ?               | ?               | ?               |
|               |                | 0.7.3-1&nbsp;&#x21D1; |                |                |                |
|*passt*        | 2024-08-14      | 2024-06-24      | 2024-08-14      | 2024-08-14      | 2024-08-14      |
|               | 2024-07-26&nbsp;&#x21D1; |                | 2024-07-26&nbsp;&#x21D1; | 2024-06-24&nbsp;&#x21D1; | 2024-07-26&nbsp;&#x21D1; |
|*podman*       | 5.2.1+ds1-2     | 4.9.4-1         | 5.2.1-1         | 5.2.1-1         | 5.2.1-1         |
|               | 5.0.3+ds1-5&nbsp;&#x21D1; |                | 5.2.0-1&nbsp;&#x21D1; | 5.1.2-1&nbsp;&#x21D1; | 5.2.0\~rc2-1&nbsp;&#x21D1; |
|*runc*         | 1.1.12+ds1-5    | 1.1.12-1        | 1.1.12-3        | 1.1.12-3        | 1.1.12-4        |
|               | 1.1.12+ds1-2&nbsp;&#x21D1; |                |                |                |                |
|*skopeo*       | 1.13.3+ds1-2+b2 | 1.16.0-1        | 1.16.0-1        | 1.16.0-1        | 1.16.0-1        |
|               | 1.13.3+ds1-2+b1&nbsp;&#x21D1; | 1.15.2-1&nbsp;&#x21D1; |                | 1.15.2-1&nbsp;&#x21D1; |                |
|*systemd*      | 256.5-1         | 254.16-1        | 255.10-3        | 255.10-3        | 256.4-1         |
|               | 256.4-2&nbsp;&#x21D1; |                | 255.10-1&nbsp;&#x21D1; | 255.10-1&nbsp;&#x21D1; |                |

**Comment by cevich** [comment]
Hrmmm, it looks like a bunch of tasks are still waiting for the container task to finish.  We don't want that since the new multi-arch fedora container takes >1hr to build.

**Comment by cevich** [comment]
Force-push: Fixed up commit message + simplified diff slightly.  I'm not sure why many/most of the VM build tasks aren't running in parallel w/ container-based tasks.  Maybe it's some Cirrus-CI quota/restriction.

**Comment by edsantiago** [comment]
Can you give context on what it is you noticed? All I saw was green CI

**Comment by cevich** [comment]
> Can you give context on what it is you noticed?

The commit message needed a minor tweak.  I thought I could simplify the YAML and just make everything use the &IBI_VM alias, but I forgot the nested virt stuff is important too -sigh-.

**Comment by cevich** [comment]
There, I think this should be better.  The last thing I want is to add a flake into these builds :confused:

**Comment by cevich** [comment]
Re:

> I'm not sure why many/most of the VM build tasks aren't running in parallel w/ container-based tasks.

I was [looking at the task scheduling sequence](https://cirrus-ci.com/build/5628105086205952) (green-bars on the right).  I seemed like a bunch of tasks were waiting for the new, slow container build.  It may have just been a fluke though, nothing in the dependency tree suggested the VM builds should block.

**Comment by cevich** [comment]
`Error: statfs /dev/kvm: no such file or directory`

That's the nested-virt isn't supported problem :disappointed:

**Comment by cevich** [comment]
Hrmmm, okay, taking a step back.  Let me just go back to the CI-green commit, then only fix the commit message, and add in the `enable_nested_virtualization: true`.

**Comment by cevich** [comment]
> I was looking at the task scheduling sequence (green-bars on the right).

Confirmed, this is misleading. VM builds are running concurrently with the container builds as intended.

**Comment by cevich** [comment]
I'm going to abandon this, the emulated build is just too slow.  The overall CI VM image build process is complex and lengthy enough. Nobody want more complexity and an even slower (1-1/2 hour) container build on top.

If somebody wants to take this up in the future, I'd suggest doing a native arm64 build, then (somehow) combining the two into a manifest-list after-the-fact.  This is also complex, but at least it'll run quickly (like 15m probably).

**Pull request closed without merging**